src/test/java/ludogorie_soft/reservations_platform_api/service/impl/BookingServiceImplTest.java

```diff
@@ -288,4 +311,77 @@ void testEditBookingShouldThrowWhenTheDatesAreNotAvailable() throws IOException,
         assertThrows(NotAvailableDatesException.class, () -> bookingService.editBooking(createdBooking.getId(), editRequestDto));
     }
 
+    @Test
+    public void testAddCustomerDataToBooking_CustomerExists() {
+        // GIVEN
+        Booking booking = BookingTestHelper.createBooking();
+        Customer existingCustomer = CustomerTestHelper.createCustomer();
+        ConfirmationToken confirmationToken = ConfirmationTokenTestHelper.createConfirmationToken();
+        BookingRequestCustomerDataDto customerData = BookingTestHelper.createBookingRequestWithCustomerData(booking, existingCustomer);
+
+        when(bookingRepository.findById(customerData.getBookingId())).thenReturn(Optional.of(booking));
+        when(customerRepository.findByFirstNameAndLastNameAndEmailAndPhoneNumber(
+                customerData.getFirstName(), customerData.getLastName(), customerData.getEmail(), customerData.getPhoneNumber()))
+                .thenReturn(Optional.of(existingCustomer));
+        when(confirmationTokenService.createConfirmationToken()).thenReturn(confirmationToken);
+
+        // WHEN
+        BookingResponseWithCustomerDataDto response = bookingService.addCustomerDataToBooking(customerData);
+
+        // THEN
+        assertNotNull(response);
+        assertEquals(booking.getId(), response.getBookingId());
```

Comment:
You can try to campare the 2 objects `usingRecursiveComparison().ignoringFields(...)` https://www.baeldung.com/java-assert-equality-no-equals

Reply:
I think using this method will require longer code to ignore all the not relative lines, than the current implementation